byterun/caml/misc.h

```diff
@@ -124,7 +124,94 @@ CAMLnoreturn_end;
 CAMLextern char * caml_strdup(const char * s);
 CAMLextern char * caml_strconcat(int n, ...); /* n args of const char * type */
 
-/* <private> */
+/* Use macros for some system calls being called from OCaml itself.
+  These calls can be either traced for security reasons, or changed to
+  virtualize the program. */
+
+#ifndef CAML_WITH_CPLUGINS
+
```

Comment:
System-call interception can be used for other things than just security. If you look at the current cplugins repository on: https://github.com/OCamlPro/ocp-cplugins , you will see that we use it for example to monitor a build, i.e. to know all the OCaml commands that are called during a build, whatever the build system used. It can be particularly useful to know the `ocamlc` commands used when a mix of `ocamlbuild`+`ocamlfind` is used... and it will also work under Windows, but requires "some" system-call interception.